### PR TITLE
User header mobile, Transaction operations, Language switcing

### PR DESCRIPTION
**Pull request by ahmedaftab-RT**
### #87  Language Selectors

- Added flags and language switching functionality
- Selected language is disabled
- Selected language code is showing with the globe icon 
**Known Issue:**
Language switching changes appearing after the refresh 
language

### #86 Transactions Operations Cards

- cards and tabular view toggling
- Operations showing dynamically in the cards 
- Field mapping for nested objects and unmapped fields in cards and tabular view
- Responsiveness
- PreAsset fileds calculations in transaction operation
all-transactions-cards)
user-cards

### #90 User Pofile Header Mobile view #90

- User profile responsive view for the mobiles

user-mob-header

### CODE REVIEW

File: src/common/api/urls.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { ConfigItems } from "../../../config";
 // Get witness Account
 export const getAccount = async (user: string) => {
   const account_url = `${ConfigItems.baseUrl}/api/get_accounts?names=["${user}"]`;
+  console.log(account_url)
```
Comment (feruzm) [inline src/common/api/urls.tsx:7]: Clear all console logs @ahmedaftab-RT

File: src/common/components/profile/UserTransactionCards.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+import { UserTransactionType } from "../../pages/profile/userTypes";
+import TransactionOperationTable from "../../pages/profile/userOpTable";
+import SpinnerEffect from "../loader/spinner";
+import { TimestampField } from "../fields/blockFields/DateTimeTable";
+import { AscendingIcon, DescendingIcon } from "../../img/svg";
+import { getUserTransaction } from "../../api/urls";
+import UserAvatar from "../user-avatar";
+import { LinkAccount } from "../../pages/fields/common_fields";
+import parseAsset from "../../helper/parse-asset";
+import { Card, Col, Row } from "react-bootstrap";
```
Comment (feruzm) [inline src/common/components/profile/UserTransactionCards.tsx:31]: move package imports at top

File: src/common/components/profile/userTransactionTable.tsx [hunk 3]
```diff
@@ -24,6 +24,7 @@ import SpinnerEffect from "../loader/spinner";
 import { TimestampField } from "../fields/blockFields/DateTimeTable";
 import { AscendingIcon, DescendingIcon } from "../../img/svg";
 import { getUserTransaction } from "../../api/urls";
+import { Col, Row } from "react-bootstrap";
```
Comment (feruzm) [inline src/common/components/profile/userTransactionTable.tsx:27]: move package imports to top

File: src/common/components/profile/UserTransactionCards.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
+import React, { useState, useEffect } from "react";
```
Comment (feruzm) [inline src/common/components/profile/UserTransactionCards.tsx:1]: component file names should start with lowercase always to keep consistency

File: src/common/components/user-header/index.tsx [hunk 5]
```diff
@@ -64,6 +64,17 @@ const UserHeader = (props: any) => {
...
             </Row>
             <Row className="user-profile-feature-row">
+              <Col md={12} className="user-header-info" >
+              <div>
+                    <h3 className="text-capitalize mh-0">
+                      {metaProfile.name ? metaProfile.name : name}
+                    </h3>
+                    <h6 className="text-capitalize">ID: {id}</h6>
+                    {metaProfile.about && (
+                      <h6 className="text-capitalize">About: {metaProfile.about}</h6>
```
Comment (feruzm) [inline src/common/components/user-header/index.tsx:74]: all text should be added into translation string

File: src/common/i18n/locales/es-ES.json [hunk 6]
```diff
@@ -5,21 +5,21 @@
   },
   "nav": {
```
Comment (feruzm) [inline src/common/i18n/locales/es-ES.json:6]: Don't edit other language files, only en-US.json. If you want to modify some strings, do them in crowdin.com, ask for invite so I can add you there.

File: src/common/pages/header/index.tsx [hunk 7]
```diff
@@ -10,6 +10,9 @@ import { brightnessSvg, hiveLogo, globeImg } from "../../img/svg";
 import { languages } from "../../languages";
 import { toggleTheme } from "../../store/global/index";
 import { _t } from "../../i18n";
+import * as ls from "../../util/local-storage";
+import i18n from "i18next";
```
Comment (feruzm) [inline src/common/pages/header/index.tsx:14]: move package imports to top

File: src/common/pages/profile/userOpTable.tsx [hunk 8]
```diff
@@ -8,6 +8,42 @@ import { AuthorityObject } from "../../components/profile/userAuthorities";
 import { DecodeJson } from "../../../server/util";
 import { LinkAccount, ObjectFieldArray, StringFieldArray } from "../fields/common_fields";
 import { UserAvatar } from "../../components/user-avatar";
+import parseAsset from "../../helper/parse-asset";
+import { isInteger, isNumber } from "lodash";
```
Comment (feruzm) [inline src/common/pages/profile/userOpTable.tsx:12]: move package imports to top

File: src/common/pages/profile/userOpTable.tsx [hunk 9]
```diff
@@ -102,6 +107,7 @@ const TransactionOperationTable = (props: any) => {
                       <tbody>
                         {Object.keys(opVal).map((key, k: number) => {
                           k = k + Math.floor(Math.random() * 10000) + 9000;
```
Comment (feruzm) [inline src/common/pages/profile/userOpTable.tsx:109]: don't use randomness for keys, use unique values from array/object in equation.

File: src/common/pages/transactions/index.tsx [hunk 10]
```diff
@@ -10,13 +10,20 @@ import SpinnerEffect from "../../components/loader/spinner";
 import BackToTopButton from "../../components/Buttons/BackToTop";
 import { getHeadBlock, getTransactions } from "../../api/urls";
 import { setHeadBlockData } from "../../store/HeadBlock";
+import { cardViewSVG, tableViewSVG } from "../../img/svg";
+import { Button, Container } from "react-bootstrap";
```
Comment (feruzm) [inline src/common/pages/transactions/index.tsx:14]: move package imports to top

File: src/common/i18n/locales/es-ES.json (hunk 6 above)
Comment (ahmedaftab-RT) [reply src/common/i18n/locales/es-ES.json:6]: Sure

File: src/common/api/urls.tsx (hunk 1 above)
Comment (ahmedaftab-RT) [reply src/common/api/urls.tsx]: Console logs have been removed

File: src/common/components/profile/UserTransactionCards.tsx (hunk 2 above)
Comment (ahmedaftab-RT) [reply src/common/components/profile/UserTransactionCards.tsx]: Updated

File: src/common/components/profile/userTransactionTable.tsx (hunk 3 above)
Comment (ahmedaftab-RT) [reply src/common/components/profile/userTransactionTable.tsx:27]: Updated

File: src/common/components/profile/UserTransactionCards.tsx (hunk 4 above)
Comment (ahmedaftab-RT) [reply src/common/components/profile/UserTransactionCards.tsx]: Updated

File: src/common/components/user-header/index.tsx (hunk 5 above)
Comment (ahmedaftab-RT) [reply src/common/components/user-header/index.tsx]: Updated in the translation files

File: src/common/i18n/locales/es-ES.json (hunk 6 above)
Comment (ahmedaftab-RT) [reply src/common/i18n/locales/es-ES.json:6]: I edited some of the keys with their respective language translation to test the localization working

File: src/common/pages/header/index.tsx (hunk 7 above)
Comment (ahmedaftab-RT) [reply src/common/pages/header/index.tsx]: Updated

File: src/common/pages/profile/userOpTable.tsx (hunk 8 above)
Comment (ahmedaftab-RT) [reply src/common/pages/profile/userOpTable.tsx]: Updated

File: src/common/pages/profile/userOpTable.tsx (hunk 9 above)
Comment (ahmedaftab-RT) [reply src/common/pages/profile/userOpTable.tsx]: All the random keys with random numbers have been resolved

File: src/common/pages/transactions/index.tsx (hunk 10 above)
Comment (ahmedaftab-RT) [reply src/common/pages/transactions/index.tsx]: Updated

File: src/common/i18n/locales/es-ES.json (hunk 6 above)
Comment (feruzm) [reply src/common/i18n/locales/es-ES.json:6]: > I edited some of the keys with their respective language translation to test the localization working

@ahmedaftab-RT Only do that from crowdin.com website so we get update from that service on separate PR.

File: src/common/routes.ts [hunk 11]
```diff
@@ -3,7 +3,7 @@ export default {
     ABOUT:'/about',
     Contact:'/contact',
     Service:'/service',
-    HeadBlock:'/head/b/:id',
+    HeadBlock:'/head/b/',
```
Comment (feruzm) [inline src/common/routes.ts:6]: @ahmedaftab-RT why headblock route has changed?

**Review by feruzm — APPROVED** [review]

**Pull request merged**